coretex/cli/utils.py

```diff
@@ -14,3 +32,58 @@ def arrowPrompt(choices: List[Any]) -> Any:
     ])
 
     return answers["option"]
+
+
+def authenticateUser(username: str, password: str) -> None:
+    response = networkManager.authenticate(username, password, False)
+
+    if response.hasFailed():
+        if response.statusCode >= 500:
+            raise RuntimeError("Something went wrong, please try again later.")
+
+        if response.statusCode >= 400:
+            raise RuntimeError("User credentials invalid, please try configuring them again.")
+
+
+def initializeUserSession() -> None:
+    config = loadConfig()
+    tokenExpirationDate = decodeDate(config["tokenExpirationDate"])
+    refreshTokenExpirationDate = decodeDate(config["refreshTokenExpirationDate"])
+
+    if datetime.utcnow().replace(tzinfo = timezone.utc) < tokenExpirationDate:
+        return
+
+    if datetime.utcnow().replace(tzinfo = timezone.utc) < refreshTokenExpirationDate:
+        refreshToken = config["refreshToken"]
+        response = networkManager.authenticateWithRefreshToken(refreshToken)
+        if response.hasFailed():
+            if response.statusCode >= 500:
+                raise RuntimeError("Something went wrong, please try again later.")
+
+            if response.statusCode >= 400:
+                authenticateUser(config["username"], config["password"])
+    else:
+        authenticateUser(config["username"], config["password"])
+
+    jsonResponse = response.getJson(dict)
+    config["token"] = jsonResponse["token"]
+    config["tokenExpirationDate"] = jsonResponse["expires_on"]
+    saveConfig(config)
```

Comment:
If `authenticateUser` was executed refresh token values must be updated too since on authentification a new refresh token will be received